### PR TITLE
PR: Fix tooltip position after the editor is undocked

### LINKED ISSUE #11355: Editor: wrong tooltip position when undocking
The tooltips are being placed incorrectly after a docking or undocking of the Editor:

tooltipe

As discussed with @ccordoba12

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

tooltip

### Issue(s) Resolved

Fixes #11355 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/widgets/calltip.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def __init__(self, parent=None, as_tooltip=False):
         """
         Shows tooltips that can be styled with the different themes.
         """
-        super(ToolTipWidget, self).__init__(parent, Qt.ToolTip)
+        super(ToolTipWidget, self).__init__(None, Qt.ToolTip)
```
Comment (goanpeca) [inline spyder/widgets/calltip.py]: Why is this needed?

File: spyder/widgets/mixins.py [hunk 2]
```diff
@@ -661,6 +661,10 @@ def hide_tooltip(self):
         self._last_point = None
         self.tooltip_widget.hide()
 
+    def reset_tooltip_tip(self):
```
Comment (goanpeca) [inline spyder/widgets/mixins.py]: `reset_tooltip_tip`?

Is there any other kind of reset, could we just use `reset_tooltip`?

File: spyder/plugins/editor/widgets/editor.py [hunk 3]
```diff
@@ -2395,6 +2395,11 @@ def __modify_stack_title(self):
             state = finfo.editor.document().isModified()
             self.set_stack_title(index, state)
 
+    def __reset_tooltip_tip(self):
```
Comment (ccordoba12) [inline spyder/plugins/editor/widgets/editor.py]: Let's change this to name to `__reset_tooltip`

File: spyder/widgets/calltip.py (hunk 1 above)
Comment (ccordoba12) [inline spyder/widgets/calltip.py]: Why does the parent of this widget need to be None?

@goanpeca, could this have other undesired side effects?

File: spyder/widgets/mixins.py (hunk 2 above)
Comment (ccordoba12) [inline spyder/widgets/mixins.py]: Let's change this name to `reset_tooltip`

File: spyder/widgets/calltip.py (hunk 1 above)
Comment (dalthviz) [reply spyder/widgets/calltip.py]: Without this the tooltips position sometimes is incorrectly set. While testing, this was the way I found to make the position calculation to work correctly. I think that maybe the `Qt.ToolTip` window flag has something to do with this behavior too but I'm unsure.

Also, I choose to try this because the `CallTipWidget` does the same thing ([sets the parent to `None` when calling the super `__init__`](https://github.com/spyder-ide/spyder/blob/5b9d1e98e54c4a44f18f454d6f1169e7a57e49e3/spyder/widgets/calltip.py#L257))
```

File: spyder/widgets/mixins.py (hunk 2 above)
Comment (dalthviz) [reply spyder/widgets/mixins.py]: Yep `reset_tooltip` everywhere sounds better. I will change it :+1:

File: spyder/widgets/calltip.py (hunk 1 above)
Comment (dalthviz) [reply spyder/widgets/calltip.py]: @ccordoba12 please see the comment above

Comment (ccordoba12) [reply spyder/widgets/calltip.py]: Ok, thanks @dalthviz for the explanation. If we're already doing that for `CallTipWidget`, then we're on the safe side doing this here too.
